### PR TITLE
Support joining a game

### PR DESCRIPTION
**Pull request by HThanki**
- [x] Add join game modal link to index page
- [x] Have the action trigger the games update method and update black_player_id with current user 
- [x] Redirect to the game you join
- [ ] Display the users stats on the Join Modal (To be done at a later date now)
- [ ] Display the users avatar on the Join Modal (To be done at a later date now)

### CODE REVIEW

File: app/views/games/index.html.erb [hunk 1]
```diff
@@ -4,7 +4,8 @@
 <% @games.each do |game| %>
   <% if game.open_seat? %>
     <p>
-      <%= link_to game.name, game_path(game) %>
+      <%= link_to game.name, game_path(game) %> -
+      <%= link_to 'Join Game', '#' %>
```
Comment (woodbrettm) [inline app/views/games/index.html.erb]: Only need the link_to game name on line 7 :)

File: app/views/games/index.html.erb [hunk 2]
```diff
@@ -4,7 +4,8 @@
 <% @games.each do |game| %>
   <% if game.open_seat? %>
```
Comment (woodbrettm) [inline app/views/games/index.html.erb]: The backend returns only open seat games so you won't need this if statement.

Comment (ghost) [inline app/views/games/index.html.erb]: How so?

Comment (woodbrettm) [inline app/views/games/index.html.erb]: On the is_available branch, I have it so the method open_games in the model returns 20 games where white_player is nil. In the def index in the controller, I have @games = Game.open_seats :)

Comment (ghost) [inline app/views/games/index.html.erb]: That branch was pushed to master? There is no open_seats method available. Maybe you accidentally pulled it out in the merge? And if thats a method that just returns true or false, its good practice to have a (?) on the end like so:  .open_seats?  It just makes it visually easier to see.

Comment (woodbrettm) [inline app/views/games/index.html.erb]: Check the master branch on github :). Oh wow LOL I just noticed a mistake I made in the games controller. In the games model, it's self.list_available_games, but the controller is calling open_seats. Should be @game = Game.list_available_games. Forgot to change D:

**Comment by mscoutermarsh** [comment]
PR looks great.

If you want, you could split out the remaining two Stats & Avatar tasks into another pull request.

**Comment by HThanki** [comment]
@It-is-Alex Current push reviewed. All looks good. Good job :+1:

**Pull request merged**